async-nats/src/lib.rs

```diff
@@ -725,15 +735,38 @@ impl Client {
         format!("_INBOX.{}", nuid::next())
     }
 
+    pub async fn queue_subscribe(
+        &mut self,
+        subject: String,
```

Comment:
I didn't want to stray from your existing conventions, but it would be slightly more ergonomic for the signature of subscribe to be 
`fn subscribe<T:ToString>(&self, subject: T)`
and for queue_subscribe to be 
`fn queue_subscribe<T1:ToString,T2:ToString>(&self, subject: T1, queue_group: T2)`

Or, if they are to be unified into a single subscribe function, then
`fn subscribe<T1:ToString,T2:ToString>(&self, subject: T1, queue_group: Option<T2>)`

If you're ok with either of those, I can also implement it on the next commit.

Reply:
`ToString` allocates, even if you pass `String`. 

`Into<String>` is a better option here.

We refrained from having those helpers to allow ourselves to decide later.
Both @caspervonb  and I want to be very clear in the codebase - especially in `publish` which can happen thousands of times per second -  what is allocated etc.

Reply:
I'm hesitant towards doing this because the `ToString` implementation for String is a deep clone, it could be `Into<String>` which should be a move but personally I'm more for explicit conversions than implicit ones as the caller is in control of how the conversion happens.

We're considering if we want a (potentially interned) type for `Subject`s, header names will likely get one in the form of `HeaderName`s from the http crate's API.

I'd hold off on this for now, at-least until the subject-type matter is resolved.

Reply:
Short notice: So far #222 is ready for review and completely integrated the `Subject` type and a `AsSubject` trait into the old code base. It may went unnoticed with all the async rewriting process.